libuavcan/include/libuavcan/build_config.hpp

```diff
@@ -62,6 +66,14 @@
 #    define LIBUAVCAN_EXPORT
 #endif
 
+/**
+ * Allows selecting the size of the microsecond timestamp type used by libuavcan.
+ * Valid values are 8, 4, and 2.
+ */
+#ifndef LIBUAVCAN_MICROSECOND_SIZE_BYTES
+#    define LIBUAVCAN_MICROSECOND_SIZE_BYTES 8
```

Comment:
Why 2 and 4 are valid? The point of 64-bit integers is that they never overflow, removing an (ostensibly) common source of errors in time handling logic where the value wraps over to zero. Allowing smaller integers would mean that the code will have to account for overflows, bringing additional complexity and possibly jeopardizing correctness. Do we really want this, or am I misunderstanding the intent? What do we gain by allowing the variability here?

Reply:
I had an argument for why I did this but I don't feel like defending it right now. I'll just  remove this and if it comes up in the real world we can revisit the argument then.